src/components/ExerciseDetails.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box, Button, FormControl, InputLabel, MenuItem, Select, SelectChangeEvent, Stack } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { Exercise } from '../utils/type';
+import { exerciseStore } from '../utils/firebase';
+
+function ExerciseDetails({ exercise }: { exercise: Exercise | null }) {
+  const templateExercise: Exercise = {
+    type: '',
+    endTime: '',
+    startTime: '',
+    id: null,
+    reps: [],
+  };
+  const workoutOptions = ['', 'pull up', 'push bar', 'squats', 'abs', 'legs'];
+
+  // const repOptions = ['', 1, 2, 3, 5, 7, 10, 12, 15, 20, 30, 45, 70, 100];
+  // const [exerciseType, setexerciseType] = useState('');
+  // const [reps, setReps] = useState(null);
+  const [exerciseData, setExerciseData] = useState(() => exercise || templateExercise);
+
+  useEffect(() => {
+    setExerciseData((prevEx) => exercise || prevEx);
+  }, [exercise]);
+
+  const showWorkoutOptions = (workoutOptions: Array<number | string>) =>
+    workoutOptions.map((workout) => (
+      <MenuItem value={workout} key={workout}>
+        {workout}
+      </MenuItem>
+    ));
+
+  const makeChangeHandler = (event: SelectChangeEvent) => {
+    setExerciseData({ ...exerciseData, type: `${event.target.value}` });
+  };
+
+  const handleUpdateorCreate = (exercise: Exercise) => {
+    exerciseStore().createOrUpdate(exercise);
+  };
+
+  const makeSubmitHandler = (event: React.FormEvent) => {
```

Comment:
Try to name your methods consistently... I'm seeing you use `make` in some places and `handle` in others.